src/pages/transactions/edit/Edit.tsx

```diff
@@ -0,0 +1,231 @@
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { FormEvent, useEffect } from 'react';
+import { Card, Element } from '@invoiceninja/cards';
+import { InputField, SelectField } from '@invoiceninja/forms';
+import { transactionTypes } from 'common/constants/transactions';
+import { useCurrencies } from 'common/hooks/useCurrencies';
+import { useTitle } from 'common/hooks/useTitle';
+import { endpoint } from 'common/helpers';
+import {
+  TransactionInput,
+  TransactionResponse,
+} from 'common/interfaces/transactions';
+import { Container } from 'components/Container';
+import { DebouncedCombobox } from 'components/forms/DebouncedCombobox';
+import { Default } from 'components/layouts/Default';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { TransactionValidation } from '../common/validation/ValidationInterface';
+import { request } from 'common/helpers/request';
+import { useNavigate, useParams } from 'react-router-dom';
+import { route } from 'common/helpers/route';
+import { toast } from 'common/helpers/toast/toast';
+import { AxiosError } from 'axios';
+import { useTransactionQuery } from '../common/queries';
+import { DecimalNumberInput } from 'components/forms/DecimalNumberInput';
+import { DecimalInputSeparators } from 'common/interfaces/decimal-number-input-separators';
+import { useResolveCurrency } from 'common/hooks/useResolveCurrency';
+
+export function Edit() {
+  const { t } = useTranslation();
+
+  const navigate = useNavigate();
+
+  const { id } = useParams<string>();
+
+  const currencies = useCurrencies();
+
+  const resolveCurrency = useResolveCurrency();
+
+  const { documentTitle } = useTitle('edit_transaction');
+
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+
+  const [errors, setErrors] = useState<TransactionValidation>();
+
+  const { data: response } = useTransactionQuery({ id });
+
+  const [transaction, setTransaction] = useState<TransactionInput>();
+
+  const [currencySeparators, setCurrencySeparators] = useState<
+    DecimalInputSeparators | undefined
+  >();
+
+  const pages = [
+    { name: t('transactions'), href: '/transactions' },
+    { name: t('edit_transaction'), href: `/transactions/${id}/edit` },
+  ];
+
+  const getCurrencySeparators = (currencyId: string) => {
+    const currency = resolveCurrency(currencyId) || currencies[0];
+    return {
+      decimalSeparator: currency?.decimal_separator,
+      precision: currency?.precision,
+      thousandSeparator: currency?.thousand_separator,
+    };
+  };
+
+  const getTransactionInputObject = (
+    responseDetails: TransactionResponse | undefined
+  ) => {
+    const { date, amount, currency_id, bank_integration_id, description } =
+      responseDetails || {};
+    setCurrencySeparators(getCurrencySeparators(currency_id || ''));
+    return {
+      base_type:
+        responseDetails?.base_type === 'CREDIT' ? 'deposit' : 'withdrawal',
```

Comment:
Please extract this to enum reference, as previously mentioned.

Reply:
> Please extract this to enum reference, as previously mentioned.

done